Lib/inspect.py

```diff
@@ -2492,6 +2494,15 @@ def _signature_from_function(cls, func, skip_bound_arg=True,
                __validate_parameters__=is_duck_function)
 
 
+def _descriptor_get(descriptor, obj):
```

Comment:
Would it make sense to move this function closer to `_signature_get_user_defined_method`? Or do you think it belongs here better?

Reply:
I added it close to `_signature_from_callable()` where it was used. Now it is also used in `_signature_get_user_defined_method()` which I planned to remove. If move it close to `_signature_get_user_defined_method()`, it will be far from `_signature_from_callable()`.

Reply:
You plan to move `_signature_get_user_defined_method()` or the usage of `_descriptor_get` in `_signature_get_user_defined_method()`? I was expecting for removal of `_descriptor_get` in `_signature_from_callable()` but maybe we are look at different paths.

Reply:
I plan to remove `_signature_get_user_defined_method()`. I only left it to minimize the diff.

Reply:
How do you plan to replace all the `_signature_get_user_defined_method()` calls for `__call__`, `__init__` and `__new__`? Are you going to do a

```python
meth = getattr(cls, '__new__', None)
meth = _descriptor_get(meth, cls)
```

when the `_NonUserDefinedCallables` issue is solved?